File: EASSET.Web/Modules/MasterData/SubUnit/SubUnitColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace EASSET.MasterData.Columns;

[ColumnsScript("MasterData.SubUnit")]
[BasedOnRow(typeof(SubUnitRow), CheckNames = true)]
public class SubUnitColumns
{
    //[EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    //public string SubUnitId { get; set; }
    [EditLink]
    public string SubUnitNama { get; set; }
    public string SubUnitKode { get; set; }
    public string UnitNama { get; set; }
    //public string ThAng { get; set; }
    //public string Op { get; set; }
    //public DateTime Lu { get; set; }
    //public string Pc { get; set; }
    //public bool Dlt { get; set; }
    public string Alamat { get; set; }
    public string Npwp { get; set; }
}

File: EASSET.Web/Modules/ServerTypes/MasterData/SubUnitColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SubUnitRow } from "./SubUnitRow";

export interface SubUnitColumns {
    SubUnitNama: Column<SubUnitRow>;
    SubUnitKode: Column<SubUnitRow>;
    UnitNama: Column<SubUnitRow>;
    Alamat: Column<SubUnitRow>;
    Npwp: Column<SubUnitRow>;
}

export class SubUnitColumns extends ColumnsBase<SubUnitRow> {
    static readonly columnsKey = 'MasterData.SubUnit';
    static readonly Fields = fieldsProxy<SubUnitColumns>();
}